src/ExpressionParser/ExpressionParsers.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\ExpressionParser;
+
+/**
+ * @template-implements \IteratorAggregate<ExpressionParserInterface>
+ *
+ * @internal
+ */
+final class ExpressionParsers implements \IteratorAggregate
+{
+    /**
+     * @var array<class-string<ExpressionParserInterface>, array<string, ExpressionParserInterface>>
+     */
+    private array $parsersByName = [];
+
+    /**
+     * @var array<class-string<ExpressionParserInterface>, ExpressionParserInterface>
+     */
+    private array $parsersByClass = [];
+
+    /**
+     * @var \WeakMap<ExpressionParserInterface, array<ExpressionParserInterface>>|null
+     */
+    private ?\WeakMap $precedenceChanges = null;
+
+    /**
+     * @param array<ExpressionParserInterface> $parsers
```

Comment:
list instead of array everywhere?

Reply:
As your own code does not care about the keys, I don't see the reason to force the caller to provide a list.